src/fsharp/service/FSharpCheckerResults.fs

```diff
@@ -850,7 +850,12 @@ type internal TypeCheckInfo
             | Some(CompletionContext.OpenDeclaration) ->
                 GetDeclaredItems (parseResultsOpt, lineStr, origLongIdentOpt, colAtEndOfNamesAndResidue, residueOpt, lastDotPos, line, loc, filterCtors, resolveOverloads, hasTextChangedSinceLastTypecheck, false, getAllSymbols)
                 |> Option.map (fun (items, denv, m) ->
-                    items |> List.filter (fun x -> match x.Item with Item.ModuleOrNamespaces _ -> true | _ -> false), denv, m)
+                    items 
+                    |> List.filter (fun x ->
+                        match x.Item with
+                        | Item.ModuleOrNamespaces _ -> true
+                        | Item.Types (_, tcrefs) when tcrefs |> List.exists (fun ty -> isAppTy g ty && isStaticClass g (tcrefOfAppTy g ty)) -> true
```

Comment:
Not sure if this is actionable, so feel free to ignore...

This is the correct change for the feature given how this routine is written, but I wonder if we should be thinking about refactoring things like this throughout the service/FSharpCheckerResults layer. Embedding details about validity like this makes it challenging (for me, at least), to make sense of this overall routine. The parameter lists pattern even more so.

Reply:
Yeah, I agree.  The whole of this routine is poorly written and has grown with adhoc cases over time. One reason I split `service.fs` into two was to start to make progress on this.